### ISSUE TITLE
more details about training the global networks

### ISSUE THREAD

**Issue by cccusername** [junyanz/interactive-deep-colorization#49]
hi, i am confused about the input and output when you training the global networks.
I will very appreciate that if you can tell me, thank you！

**Comment by cccusername** [comment]
@richzhang @junyanz

**Comment by richzhang** [comment]
The input is the global color histogram and the grayscale channel. The output is the color channels.

**Comment by cccusername** [comment]
Could you share the training code？Thank you

**Comment by richzhang** [comment]
I'm working on a pytorch version right now. I'll probably release the local hints training first. Thanks for your patience.

**Comment by richzhang** [comment]
Local Hints network training is in this [repository](https://github.com/richzhang/colorization-pytorch/tree/master). Thanks for your patience.

**Comment by cccusername** [comment]
thanks a lot @richzhang

**Comment by Peter2021** [comment]
hi, i am confused about the loss in global hints network.

1. I can't find any details about global network's loss in this repository. When I see local hints network in pytorch version, I find huber loss were abandoned in training.
2. The other thing, is about network. I reimplement global hint network in pytorch based on this repository and I use main branch in local hints network in pytorch version. I choose huber loss or L1 loss, but I get yellowish results in visdom.
3. By the way, I use expand_as() function in pytorch to expand global hints. Is necessary tor write its derivation process?

I will very appreciate that if you can tell me, thank you！

**Comment by Peter2021** [comment]
@junyanz @richzhang about global hints network
hi, i am confused about the loss in global hints network.

I can't find any details about global network's loss in this repository. When I see local hints network in pytorch version, I find huber loss were abandoned in training.
The other thing, is about network. I reimplement global hint network in pytorch based on this repository and I use main branch in local hints network in pytorch version. I choose huber loss or L1 loss, but I get yellowish results in visdom.
By the way, I use expand_as() function in pytorch to expand global hints. Is necessary tor write its derivation process?
I will very appreciate that if you can tell me, thank you！
